src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -1783,6 +1906,17 @@ static void glb_update_locked(grpc_exec_ctx *exec_ctx, grpc_lb_policy *policy,
         &glb_policy->lb_channel_connectivity,
         &glb_policy->lb_channel_on_connectivity_changed, NULL);
   }
+
+  // Propagate update to \a fallback_backend_addresses if the fallback is still 
+  // necessary, which is indicated by a non-NULL \a fallback_backend_addresses.
+  if (glb_policy->fallback_backend_addresses != NULL) {
```

Comment:
Do we ever use the `expiration_interval` field in a server list? I don't find any. And we mentioned this in the GSLB meeting? Anyways, I feel it's like a time bomb for the fallback logic, because it may cause the `serverlist` to be NULL from some non-NULL value. 

That's  why I prefer using `fallback_backend_address != NULL` to check, which is stable and contained in the fallback logic itself, if we only want fallback in the beginning of the connection.

Reply:
We do not currently ever set `serverlist` to NULL once we originally set it, and I think it's safe to rely on that.  We might need to change this in the future if we decide to also start using the fallback if we can't talk to any of the backends indicated in the serverlist, but for now, the code guarantees this.

To say this another way: Currently, we are structuring the code such that only one of `serverlist` and `fallback_backend_addresses` will ever be non-NULL.  As a result, I think we should use the same criteria everywhere for determining which of those two fields we should update.

Reply:
So I guess we currently don't use the expiration information for the servers. Then using `serverlist` to determine the status for fallback is also good. Maybe I am being paranoid, but I feel it's still slightly riskier, because if `expiration_interval` is used in the future, we need to care about changing the fallback code to not use fallback unexpectedly. Admittedly, using `serverlist` to determine the fallback status is better if we do want to fall back when the number of servers drops to zero, which is not our target behavior for now. 

Yes, I totally agree that we should use a single criteria to determine whether we still need to update the fallback (I say "whether we still need to update the fallback" because we always update `serverlist`, but we don't always update the fallback because it's not preferred). Either `serverlist` or `fallback_backend_addresses` can correctly achieve the goal. But we have some different opinions about which one is better (a little bit).

I will leave my reasoning in an other comment.